src/components/AnonymousChat/child/ChatItemMyTextV2.tsx

```diff
@@ -115,55 +51,140 @@ const ChatItemMyTextV2 = ({
 
-  const handleTextContainerStyle = () => {
-    if (chatType === SIGNED) {
-      return [styles.containerSigned, styles.textContainer];
-    }
-    return [styles.containerAnon, styles.textContainer];
-  };
+  const onSwipeToReply = React.useCallback(
+    (direction: 'left' | 'right') => {
+      if (direction === 'right') return;
+      if (swipeableRef.current) swipeableRef.current?.close();
+      setReplyPreview({
+        id: data?.id ?? data?.message?.id,
+        user: {username},
+        message,
+        message_type: messageType as MessageType,
+        updated_at: time,
+        chatType
+      });
+    },
+    [data, username, time, message, messageType, chatType]
+  );
 
   return (
-    <View style={styles.chatContainer}>
-      <View style={handleTextContainerStyle()}>
-        {!isContinuous && (
-          <View style={styles.chatTitleContainer}>
-            <Text style={styles.userText}>{username}</Text>
-            <View style={styles.dot} />
-            <Text style={styles.timeText}>{time}</Text>
-          </View>
-        )}
-        <Text ref={messageRef} style={styles.text}>
-          {`${message}`}
-        </Text>
-
-        {renderIcon()}
+    <Swipeable
+      testID="swipeable"
+      ref={swipeableRef}
+      friction={3}
+      overshootFriction={2}
+      onSwipeableOpen={onSwipeToReply}
+      renderLeftActions={replyIcon}>
+      <View style={containerStyle(true, isReplyPrompt)}>
+        <Animated.View
+          style={[
+            styles.wrapper,
+            animatedBubbleStyle as ViewStyle,
+            animatedPulseStyle as ViewStyle
+          ]}>
+          <ContextMenu
+            previewBackgroundColor="transparent"
+            style={{flex: 1}}
+            actions={contextMenuActions}
+            onPress={(e) => onContextMenuPressed(e, data, chatType)}>
+            <View style={styles.radius8}>
+              <View style={textContainerStyle(true, chatType, isNewLine)}>
+                {isShowUserInfo && (
+                  <View style={styles.chatTitleContainer}>
+                    <Text style={[styles.userText, textStyle(true)]}>{username}</Text>
+                    <View style={dotStyle(true)} />
+                    <Text testID="timestamp" style={[styles.timeText, textStyle(true)]}>
+                      {calculateTime(time, true)}
+                    </Text>
+                  </View>
+                )}
+
+                <ChatReplyView
+                  type={chatType}
+                  messageType={messageType}
+                  replyData={data?.reply_data}
+                />
+
+                <Text style={messageStyle(true, messageType)} onTextLayout={onTextLayout}>
+                  {`${message}`}
+                </Text>
+
+                {renderIcon()}
+              </View>
+            </View>
+          </ContextMenu>
+
+          {renderAvatar()}
+        </Animated.View>
       </View>
-      {renderAvatar()}
-    </View>
+    </Swipeable>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of the `Swipeable` component and context menu is well implemented. However, I noticed that you're not handling potential null values for `swipeableRef.current` in your `onSwipeToReply` function. Consider adding a check for this to prevent possible runtime errors.

```diff
  const onSwipeToReply = React.useCallback(
    (direction: 'left' | 'right') => {
      if (direction === 'right') return;
+     if (swipeableRef.current) {
        swipeableRef.current?.close();
        setReplyPreview({
          id: data?.id ?? data?.message?.id,
          user: {username},
          message,
          message_type: messageType as MessageType,
          updated_at: time,
          chatType
        });
+     }
    },
    [data, username, time, message, messageType, chatType]
  );
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->